src/Microsoft.AspNet.Mvc.Core/Formatters/JsonPatchInputFormatter.cs

```diff
@@ -1,126 +1,50 @@
-
-            _jsonSerializerSettings = new JsonSerializerSettings
-            {
-                MissingMemberHandling = MissingMemberHandling.Ignore,
-
-                // Limit the object graph we'll consume to a fixed depth. This prevents stackoverflow exceptions
-                // from deserialization errors that might occur from deeply nested objects.
-                MaxDepth = DefaultMaxDepth,
+            // Remove json mediatype header and only include json-patch+json value.
+            SupportedMediaTypes.Remove(MediaTypeHeaderValue.Parse("application/json"));
+            SupportedMediaTypes.Remove(MediaTypeHeaderValue.Parse("text/json"));
 
-                // Do not change this setting
-                // Setting this to None prevents Json.NET from loading malicious, unsafe, or security-sensitive types
-                TypeNameHandling = TypeNameHandling.None
-            };
-
-            _jsonSerializerSettings.ContractResolver = new JsonContractResolver();
+            SupportedMediaTypes.Add(MediaTypeHeaderValue.Parse("application/json-patch+json"));
         }
 
-        /// <summary>
-        /// Gets or sets the <see cref="JsonSerializerSettings"/> used to configure the <see cref="JsonSerializer"/>.
-        /// </summary>
-        public JsonSerializerSettings SerializerSettings
+        /// <inheritdoc />
+        public override Task<object> ReadRequestBodyAsync([NotNull] InputFormatterContext context)
         {
-            get { return _jsonSerializerSettings; }
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
+            var jsonSerializer = CreateJsonSerializer();
+            var jsonPatchDocument = base.ReadRequestBodyAsync(context).Result as IJsonPatchDocument;
+            jsonPatchDocument.ContractResolver = jsonSerializer.ContractResolver;
 
-                _jsonSerializerSettings = value;
-            }
+            return Task.FromResult((object)jsonPatchDocument);
         }
 
         /// <inheritdoc />
-        public override Task<object> ReadRequestBodyAsync([NotNull] InputFormatterContext context)
+        public override bool CanRead(InputFormatterContext context)
         {
-            var type = context.ModelType;
-            var request = context.ActionContext.HttpContext.Request;
-            MediaTypeHeaderValue requestContentType = null;
-            MediaTypeHeaderValue.TryParse(request.ContentType, out requestContentType);
-
-            // Get the character encoding for the content
-            // Never non-null since SelectCharacterEncoding() throws in error / not found scenarios
-            var effectiveEncoding = SelectCharacterEncoding(requestContentType);
-
-            using (var jsonReader = CreateJsonReader(context, request.Body, effectiveEncoding))
+            if (!typeof(IJsonPatchDocument).IsAssignableFrom(context.ModelType))
```

Comment:
Have you tested the formatter where the declared parameter type is `IJsonPatchDocument`? what happens?